keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -268,96 +193,34 @@ To control which driver is allowed to be used within a pod spec, this design wil
 * The Kubelet will create necessary mount point paths
-* Kubelet will auto-generate a volumeHandle based on `podUID` and `pod.spec.volume[x].name` (see above for detail)
-* Ephemeral drivers will receive mount-like calls (NodePublish) with generated paths and volumeHandle
-* Ephemeral drivers are responsible for handling volume operations (create, mount, unmount, delete) during this stage
-
-## Security considerations
-As written above, external attacher may require permissions to read Secrets in any namespace. It is up to CSI driver author to document if the driver needs such permission (i.e. access to Secrets at attach/detach time) and up to cluster admin to deploy the driver with these permissions or restrict external attacher to access secrets only in some namespaces.
-
-* Since access to in-line volumes can be configured by `PodSecurityPolicy` (see above), we expect that cluster admin gives access to CSI drivers that require secrets at detach time only to educated users that know they should not delete Secrets used in volumes.
-* Number of CSI drivers that require Secrets on detach is probably very limited. No in-tree Kubernetes volume plugin requires them on detach.
-* We will provide clear documentation that using in-line volumes drivers that require credentials on detach may leave orphaned attached volumes that Kubernetes is not able to detach. It's up to the cluster admin to decide if using such CSI driver is worth it.
+* Kubelet will auto-generate a volumeHandle based on `podUID` and `pod.spec.volume[x].name` (see above for detail).
+* CSI driver will receive mount-like calls (NodePublish) with generated paths and generated volumeHandle.
+
+Since ephemeral volume requests will participate in only the mount/unmount volume operation phase, CSI drivers are responsible for implementing all necessary operations during that phase (i.e. create, mount, unmount, delete, etc).  For instance, a driver would be responsible for provisioning any new volume resource during the `NodePublishStage` and for tearing down these resources during the `NodeUnpublish` call.
+
+
+## Test plans
+
+### All unit tests
+* Volume operation that use CSIVolumeSource can only work with proper feature gate enabled
+
+### Ephemeral inline volumes unit tests
+* Ensure required fields are provided
+* Mount/Unmount should be triggered with CSIVolumeSource
+* Expected generated volumeHandle is created properly
+* Ensure volumeHandle conforms to resource naming format
+* CSIVolumeSource info persists in CSI json file during mount/unmount
+* Ensure Kubelet skips attach/detach when `CSIDriver.InlineMode = true`
+* Ensure Kubelet skips inline logic when `CSIDriver.InlineMode = false` 
```

Comment:
Still a couple of references to InlineMode here.

Reply:
ok thanks.